message_ix/macro.py

```diff
@@ -388,22 +402,24 @@ def add_model_data(base, clone, data):
     c.read_data()
     c.derive_data()
 
+    # removing old initializeyear_macro before adding new one
+    cat = clone.set("cat_year", {"type_year": "initializeyear_macro"})
+    if not cat.empty:
+        clone.remove_set("cat_year", cat)
+
     # add temporal set structure
     clone.add_set("type_year", "initializeyear_macro")
     clone.add_set("cat_year", ["initializeyear_macro", c.init_year])
-    clone.add_set("type_year", "baseyear_macro")
-    clone.add_set("cat_year", ["baseyear_macro", c.base_year])
 
     # add nodal set structure
     clone.add_set("type_node", "economy")
     for n in c.nodes:
         clone.add_set("cat_node", ["economy", n])
 
     # add sectoral set structure
-    # TODO we shouldn't need a loop here
-    for s in c.sectors:
-        clone.add_set("sector", s)
-        clone.add_set("mapping_macro_sector", [s, s, "useful"])
+    for s, l in product(c.sectors, c.levels):
```

Comment:
I suggest modifying here to also differentiate the sectors and commodities which might not be always the same as suggested by Volker. 

```
for s, l, c in product(c.sectors, c.levels, c.commodities):
        clone.add_set('sector', s)
        clone.add_set("mapping_macro_sector", [s, c, l])
```